### PR TITLE
chore(sast): SAST integration tests (excluding Windows)

### PR DESCRIPTION
**Pull request by rotemavni**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

## Description

1. Created a new pipeline for SAST integration tests using an API key. 
2. Did some changes in setting the SAST artifact - adjusted the regex to allow having `v` in the version + invalidate the locally found artifact to fetch a new one every day.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my feature, policy, or fix is effective and works
- [x] New and existing tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: .github/workflows/pr-test.yml [hunk 1]
```diff
@@ -117,6 +117,50 @@ jobs:
         run: |
           pipenv run pytest integration_tests -k 'not api_key'
 
+  sast-integration-tests:
+    strategy:
+      fail-fast: true
+      matrix:
+        python: [ "3.7", "3.8" ]
```
Comment (gruebel) [inline .github/workflows/pr-test.yml:124]: we can probably remove `3.7` we will EOL it in `checkov` end of September

File: sast_integration_tests/prepare_data.sh [hunk 2]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+
+pipenv run checkov -s --framework sast_python -d flask --bc-api-key $BC_KEY -o json > checkov_report_sast_python.json
+pipenv run checkov -s --framework sast_java -d jenkins --bc-api-key $BC_KEY -o json > checkov_report_sast_java.json
+pipenv run checkov -s --framework sast_javascript -d axios --bc-api-key $BC_KEY -o json > checkov_report_sast_javascript.json
```
Comment (gruebel) [inline sast_integration_tests/prepare_data.sh:5]: you could remove `--bc-api-key` and just set the env var `BC_API_KEY`

**Review by gruebel — COMMENTED** [review]
I would recommend to just `BC_API_KEY` env var instead of creating a new one `BC_KEY` which removes the need to pass it as a flag

**Comment by rotemavni** [comment]
@gruebel Fixed your comments

File: sast_integration_tests/run_integration_tests.sh
Comment (gruebel) [inline sast_integration_tests/run_integration_tests.sh:1]: can you also change the `BC_KEY` usage here, thanks!

**Review by gruebel — APPROVED** [review]
🏆

**Review by achiar99 — APPROVED** [review]

**Pull request merged**